Print/src/PrimaryParticlePrinter.cc

```diff
@@ -0,0 +1,71 @@
+#include "Print/inc/PrimaryParticlePrinter.hh"
+#include "Print/inc/GenParticlePrinter.hh"
+#include "Print/inc/SimParticlePrinter.hh"
+#include "art/Framework/Principal/Provenance.h"
+#include <string>
+#include <iomanip>
+
+void 
+mu2e::PrimaryParticlePrinter::Print(art::Event const& event,
+				std::ostream& os) {
+  if(verbose()<1) return;
+  if(tags().empty()) {
+    // if a list of instances not specified, print all instances
+    std::vector< art::Handle<PrimaryParticle> > vah;
+    event.getManyByType(vah);
+    for (auto const & ah : vah) Print(ah);
+  } else {
+    // print requested instances
+    for(const auto& tag : tags() ) {
+      auto ih = event.getValidHandle<PrimaryParticle>(tag);
+      Print(ih);
+    }
+  }
+}
+
+void 
+mu2e::PrimaryParticlePrinter::Print(const art::Handle<PrimaryParticle>& handle,
+				std::ostream& os) {
+  if(verbose()<1) return;
+  // the product tags with all four fields, with underscores
+  std::string tag = handle.provenance()->productDescription().branchName();
+  tag.pop_back(); // remove trailing dot
+  PrintHeader(tag,os);
+  Print(*handle);
+}
+
+void 
+mu2e::PrimaryParticlePrinter::Print(const art::ValidHandle<PrimaryParticle>& handle,
+				std::ostream& os) {
+  if(verbose()<1) return;
+  // the product tags with all four fields, with underscores
+  std::string tag = handle.provenance()->productDescription().branchName();
+  tag.pop_back(); // remove trailing dot
+  PrintHeader(tag,os);
+  Print(*handle);
+}
+
+
+void 
+mu2e::PrimaryParticlePrinter::Print(const mu2e::PrimaryParticle& obj, int ind, std::ostream& os) {
+  if(verbose()<1) return;
+  os << std::setiosflags(std::ios::fixed | std::ios::right);
+  std::string gtag("GenParticleFrom PrimaryParticle");
+  gprint_.PrintHeader(gtag, os);
```

Comment:
Please replace this with text not from PrintHeader.  Currently there is a one-to-one correspondence between calls to PrintHeader and products, which helps with visual presentation and text manipulation.  You can just change this into a print of the string you have here, gtag, preferably indented.

Reply:
Done